packages/sirius-web/backend/sirius-web-domain/src/main/java/org/eclipse/sirius/web/domain/boundedcontexts/semanticdata/SemanticData.java

```diff
@@ -131,6 +131,15 @@ public void updateDocuments(ICause cause, Set<Document> newDocuments, Set<String
         }
     }
 
+    public void addDependencies(ICause cause, List<AggregateReference<SemanticData, UUID>> dependencySemanticDataIds) {
+        List<SemanticDataDependency> newDependencies = dependencySemanticDataIds.stream()
+            .map(SemanticDataDependency::new)
+            .toList();
+        this.dependencies.addAll(newDependencies);
+        this.lastModifiedOn = Instant.now();
+        this.registerEvent(new SemanticDataUpdatedEvent(UUID.randomUUID(), this.lastModifiedOn, cause, this));
```

Comment:
I reused this event because right now we don't really need a dedicated event

Reply:
When someone will ask for it, we will introduce events with a smaller granularity